src/test/groovy/com/example/CardServiceSpec.groovy

```diff
@@ -0,0 +1,102 @@
+package com.example
+
+import spock.lang.Specification
+
+class CardServiceSpec extends Specification {
+
+    def cardRepository = Mock(ListCardRepository)
+
+    def cardService = new CardService(cardRepository)
+
+    def "should return empty list of cards"() {
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        1 * cardRepository.findAll() >> { [] }
+        cards.size() == 0
+    }
+
+    def "should return list with one element"() {
+        given:
+        def card = new Card("foo", Collections.singletonList("bar"))
+
+        when:
+        def cards = cardService.findAll()
+
+        then:
+        1 * cardRepository.findAll() >> { [card] }
```

Comment:
Tworzenie stuba powinno być w sekcji given (po to ona jest).

I oczywiście nie sprawdzamy czy ta metoda została wywołana (chodzi o `1 *`). Patrz komentarz na górze pliku.